library/math/numbersLine/src/hooks/useHookAction.ts

```diff
@@ -0,0 +1,30 @@
+import { useNumbersLineContext } from "@/context/numbersLineContext";
+import { IElement } from "../type/moveable";
+import { v4 as uuidv4 } from "uuid";
+
+export const useAction = () => {
+  const { dragElements, setDragElements, setIdDraggElementClick } = useNumbersLineContext();
+
+  const deleteDragElement = (elementId: string) => {
+    const newDragElements = dragElements.filter((element) => element.id !== elementId);
+    setDragElements(newDragElements);
+  };
+
+  const duplicateDragElement = (elementId: string) => {
+    const elementToDuplicate = dragElements.find((element) => element.id === elementId);
+    if (elementToDuplicate) {
+      const newElement = {
+        ...elementToDuplicate,
+        id: uuidv4(),
+      };
+      const newDragElements: IElement[] = dragElements ? [...dragElements, newElement] : [];
```

Comment:
if you have element to duplicate you must have an array of dragElements. this query is unnecessary, you can send element and not id and then you won't have to find it by id and dont have to ask if it exist.